fedimint-core/src/api.rs

```diff
@@ -775,6 +775,30 @@ impl<C> WsFederationApi<C> {
     }
 }
 
+/// For a guardian to communicate with their server
+pub struct WsAuthenticatedApi {
+    inner: DynFederationApi,
+    auth: ApiAuth,
+}
+
+impl WsAuthenticatedApi {
+    pub fn new(url: Url, our_id: PeerId, auth: ApiAuth) -> Self {
+        Self {
+            inner: WsFederationApi::new(vec![(our_id, url)]).into(),
+            auth,
+        }
+    }
+
+    pub async fn signal_upgrade(&self) -> FederationResult<()> {
+        self.inner
+            .request_current_consensus(
+                "/upgrade".to_owned(),
+                ApiRequestErased::default().with_auth(&self.auth),
+            )
+            .await
+    }
+}
```

Comment:
nit: would be easier to review if these auth changes were in a separate commit from the signalling changes